pkg/daemon/controller/controller.go

```diff
@@ -65,6 +66,7 @@ type ServiceController struct {
 	ctx            context.Context
 	nodeName       string
 	watchNamespace string
+	psCmdRunner    powershell.CommandRunner
```

Comment:
Why do you need `psCmdRunner` to be part of the struct?

Reply:
The main reason is that  `CommandRunner` provides a method of interacting with an external program. In order to enable useful unit tests, this interface must be made part of the definition of the `ServiceController`. This makes sense to me in the same way that a logger or any other interface which interacts with systems outside of would be part of the struct.

Reply:
I don't buy this argument. Passing in `nil` values is just not a good pattern. Saying this is needed for unit is not good enough. It might be worth re-examining the object design here.